src/core/tests/widgets/test_base.py

```diff
@@ -56,6 +56,74 @@ def test_adding_children(self):
         self.widget.add(child1, child2)
         self.assertEqual(self.widget.children, [child1, child2])
 
+    def test_adding_child_with_existing_parent(self):
+        # Create a second parent widget.
+        widget2 = toga.Widget(factory=toga_dummy.factory)
+        # Create a child widget to add to widget2 before adding to widget.
+        child = toga.Widget(factory=toga_dummy.factory)
+
+        widget2._children = []
```

Comment:
A test should not access a protected or private attribute unless it is absolutely necessary. Is it the case here? No, since _children is already an empty set. Moreover, if it wasn't you could always set the public `children` property.

Reply:
I did try this test without setting `_children` to `[]`, but got the following error:

`File "c:\...\appdata\local\programs\python\python38-32\lib\site-packages\travertino\node.py", line 76, in add
    raise ValueError('Cannot add children')
ValueError: Cannot add children
`
I noticed that this behaviour was tested on line 40.

Attempting to set the public children property raised:

`AttributeError: can't set attribute`

I saw the comments on lines 43 and 44 about setting `_children` to `[]` for test purposes, so decided to mirror this approach.

Is there another way to resolve this issue?

Reply:
At the moment it seems that this is the right way to go. My bad!

Probably worth fixing that in Travertino, but that's a task for another time :)